sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -170,7 +173,7 @@ object FileSourceStrategy extends Strategy with Logging {
         l.resolve(fsRelation.dataSchema, fsRelation.sparkSession.sessionState.analyzer.resolver)
 
       // Partition keys are not available in the statistics of the files.
-      val dataFilters = normalizedFilters.filter(_.references.intersect(partitionSet).isEmpty)
+      val dataFilters = normalizedFiltersWithoutSubqueries.filter(_.references.intersect(partitionSet).isEmpty)
```

Comment:
Why data filters use `normalizedFiltersWithoutSubqueries` instead of `normalizedFilters`? subquery filters only apply to partition keys?

Reply:
Looks like this assumes that data filters are only used in data source pushing down filters. But `FileIndex.listFiles` also has data filters as parameter. `listFiles` doesn't clearly define how it will use...

Reply:
That would bring back https://github.com/apache/spark/pull/22518 again. If you check that PR you will see that `partitionKeyFilters` were filtered to exclude subqueries even before the PR so `dataFilters` must have caused that issue.

Reply:
Can you add a few lines of comment here and mention why we use `normalizedFiltersWithoutSubqueries`?

Reply:
Added.